src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantOverlapAnnotator.java

```diff
@@ -156,26 +158,39 @@ public VariantContext annotateOverlap(final List<VariantContext> overlapTestVCs,
     private static String getRsID(final List<VariantContext> rsIDSourceVCs, final VariantContext vcToAnnotate) {
         Utils.nonNull(rsIDSourceVCs, "rsIDSourceVCs cannot be null");
         Utils.nonNull(vcToAnnotate, "vcToAnnotate cannot be null");
+        final List<String> rsids = new ArrayList<>();
 
-        for ( final VariantContext vcComp : rsIDSourceVCs ) {
-            if ( vcComp.isFiltered() ) {
+        final List<VariantContext> vcAnnotateList = GATKVariantContextUtils.splitVariantContextToBiallelics(vcToAnnotate, true,
+                GenotypeAssignmentMethod.SET_TO_NO_CALL_NO_ANNOTATIONS, true);
+
+        for ( final VariantContext vcCompSource : rsIDSourceVCs ) {
+            if ( vcCompSource.isFiltered() ) {
                 continue; // don't process any failed VCs
             }
 
-            if ( ! vcComp.getContig().equals(vcToAnnotate.getContig()) || vcComp.getStart() != vcToAnnotate.getStart() ) {
-                throw new IllegalArgumentException("source rsID VariantContext " + vcComp + " doesn't start at the same position as vcToAnnotate " + vcToAnnotate);
+            if (!vcCompSource.getContig().equals(vcToAnnotate.getContig())) {
+                throw new IllegalArgumentException("source rsID VariantContext " + vcCompSource + " is not on same chromosome as vcToAnnotate " + vcToAnnotate);
             }
 
-            if ( vcToAnnotate.getReference().equals(vcComp.getReference()) ) {
-                for ( final Allele allele : vcToAnnotate.getAlternateAlleles() ) {
-                    if ( vcComp.getAlternateAlleles().contains(allele) ) {
-                        return vcComp.getID();
+            final List<VariantContext> vcCompList = GATKVariantContextUtils.splitVariantContextToBiallelics(vcCompSource, true,
+                    GenotypeAssignmentMethod.SET_TO_NO_CALL_NO_ANNOTATIONS, true);
+            boolean addThisID = false;
+            for (final VariantContext vcComp : vcCompList) {
+                for (final VariantContext vcToAnnotateBi : vcAnnotateList) {
+                    if (vcComp.getStart() == vcToAnnotateBi.getStart() && vcToAnnotateBi.getReference().equals(vcComp.getReference()) && vcComp.getAlternateAlleles().equals(vcToAnnotateBi.getAlternateAlleles())) {
```

Comment:
Wait a minute.... is vcToAnnotateBi.getAlternateAlleles() really adequate here? This means that two deletions of different lengths are exactly the same as far as our DBSNP annotations are concerned? That doesn't seem right...

Reply:
But it would care about insertions being an exact match?

Reply:
The reference allele is also checked, so deletions of different lengths shouldn't be considered a match.  I will add tests to confirm this though.